core/src/main/scala/kafka/admin/ConsumerGroupCommand.scala

```diff
@@ -277,17 +277,11 @@ object ConsumerGroupCommand {
 
     private def getZkConsumer(brokerId: Int): Option[SimpleConsumer] = {
       try {
-        zkUtils.readDataMaybeNull(ZkUtils.BrokerIdsPath + "/" + brokerId)._1 match {
-          case Some(brokerInfoString) =>
-            Json.parseFull(brokerInfoString) match {
-              case Some(m) =>
-                val brokerInfo = m.asInstanceOf[Map[String, Any]]
-                val host = brokerInfo.get("host").get.asInstanceOf[String]
-                val port = brokerInfo.get("port").get.asInstanceOf[Int]
-                Some(new SimpleConsumer(host, port, 10000, 100000, "ConsumerGroupCommand"))
-              case None =>
-                throw new BrokerNotAvailableException("Broker id %d does not exist".format(brokerId))
-            }
+        zkUtils.getBrokerInfo(brokerId) match {
+          case Some(brokerInfo) =>
+            Some(new SimpleConsumer(brokerInfo.getBrokerEndPoint(SecurityProtocol.PLAINTEXT).host,
+                                    brokerInfo.getBrokerEndPoint(SecurityProtocol.PLAINTEXT).port,
+                                    10000, 100000, "ConsumerGroupCommand"))
           case None =>
             throw new BrokerNotAvailableException("Broker id %d does not exist".format(brokerId))
```

Comment:
This can be written a bit more nicely along these lines:

```scala
zkUtils.getBrokerInfo(brokerId).map { brokerInfo =>
  new SimpleConsumer(...)
}.getOrElse(throw new ....)
```

Same applies to the other class.